### ISSUE TITLE
hard to figure PWD in inherited/sourced files 

### ISSUE THREAD

**Issue by gretel** [direnv/direnv#218]
the value of `PWD` seems to get changed when `direnv` finishes. therefore, i put this function in `.direnvrc` to work around:

```
# figures the current working directory without relying on PWD (which will change when direnv is done)
get_cwd() {
  p="$(expand_path "$1")"
  # http://stackoverflow.com/questions/965053/extract-filename-and-extension-in-bash#comment14090830_1403489
  local t="${p%${p##*/}}"
  echo "${t%/}"
}
```
so as long as `$1` is getting passed
```
use whatever "$1"
```
the according function can get the PWD _of_ the inherited/sourced file 
```
use_whatever() {
  local cwd
  cwd="$(get_cwd "$1")"
}
```
initially, i tried this [snippet](https://github.com/direnv/direnv/issues/98#issuecomment-33677881) but it will always read from the directory `.direnvrc` is in.
while my hack is working fine i'd be happy to have a more elegant solution. regards

**Comment by zimbatm** [comment]
The PWD is always the directory where the .envrc is living, even when it's sourced by another .envrc. The idea is that the outcome is then predictable since the environment shouldn't be reloaded if the user navigates to sub-directories.

What kind of use case requires to know the current directory or original .envrc ? In those cases I think it's best to declare a function in the parent .envrc and then call that function in the other child envrc.

**Comment by zimbatm** [comment]
The PWD is always the directory where the .envrc is living, even when it's sourced by another .envrc. The idea is that the outcome is then predictable since the environment shouldn't be reloaded if the user navigates to sub-directories.

What kind of use case requires to know the current directory or original .envrc ? In those cases I think it's best to declare a function in the parent .envrc and then call that function in the child envrc.

**Comment by gretel** [comment]
@zimbatm like sourcing files from the location of the original `.envrc`, well, some sort of inheritance. maybe just mention that conceptual thing in the documentation.
btw, i noticed the actual code and the documentation seem to be off-sync, lots of recently added features people might like lots 🐵

**Comment by zimbatm** [comment]
Feel free to submit a PR for better docs, I've become quite blind of it over time.

Thanks for mentioning, just pushed the latest doc.

**Comment by halostatue** [comment]
I’m going to bump this issue because this breaks some custom scripts that I have in my direnvrc (I’ve shared them on the wiki).

Specifically, I would expect `find_up` (or `find_up_` as I created in the [wiki](https://github.com/direnv/direnv/wiki/Find-Up-with-Alternates)) to search from the working directory of the main .envrc being sourced. This way, I could build something like the [`use_chruby`](https://github.com/direnv/direnv/wiki/Ruby#-chruby) function from my `~/.envrc` and it will pick up the `.ruby-version` file in the _actual_ current directory.

I can create this as a separate bug, but I think that even if the decision is made _not_ to change this particular behaviour, it would be useful to expose the base working directory `$BASE_PWD` in sourced directories and provide `find_up` or `find_up_` ways to work from `$BASE_PWD` instead of `$PWD` for this particular behaviour. I can work around this by adding `use chruby --auto` in my project `.envrc` file, but that seems a little…ugly.

**Comment by zimbatm** [comment]
Is your goal to automatically setup the project without having to create a new `.envrc` file per project?

With the current design direnv doesn't reload the `~/.envrc` when changing directory into a child folder.

**Comment by halostatue** [comment]
No, I’m not looking for that level of automation (I could get that just by using `chruby-auto.sh`), but rather if I _have_ a `.envrc` that includes `source_up`, I don’t necessarily need to specify `use chruby --auto` in the project `.envrc`. I found the behaviour here a little weird, and when I was going to create a new issue, this one popped up and is basically the same issue that I have.

**Comment by heygarrett** [comment]
> the environment shouldn't be reloaded if the user navigates to sub-directories

I have a different use case, but this is something I'd love to be able to do. For a given project I create worktree directories inside of a base directory:

```
project/
	main/
	feature-branch/
	review-branch/
```

Using the above as an example, I normally put a `.envrc` in the `project/` directory so I don't need to duplicate it across each worktree directory. But when I'm working in one of those worktrees I'd like to be able to use the actual cwd in `.envrc`.
